packages/nimble/inst/tests/test-dynamicIndexing.R

```diff
@@ -19,54 +19,209 @@ ans3 <- sapply(testsInvalidDynIndexValue, test_dynamic_indexing_model)
 ## check conjugacy detection
 
 test_that("Testing conjugacy detection with dynamic indexing", { 
-          code = nimbleCode({
-              for(i in 1:4) 
-                  y[i] ~ dnorm(mu[k[i]], sd = 1)
-              for(j in 1:5) 
-                  mu[j] ~ dnorm(0, 1)
-          })
-          m = nimbleModel(code, data = list(y = rnorm(4)),
-                          inits = list(k = rep(1,4)))
-          conf <- configureMCMC(m)
-          expect_match(conf$getSamplers()[[1]]$name, "conjugate_dnorm_dnorm",
-                       info = "failed to detect normal-normal conjugacy")
-          code = nimbleCode({
-              for(i in 1:4) 
-                  y[i] ~ dpois(mu[k[i]])
-              for(j in 1:5)
-                  mu[j] ~ dnorm(0, 1)
-          })
-          m = nimbleModel(code, data = list(y = rpois(4, 1)),
-                          inits = list(k = rep(1,4)))
-          conf <- configureMCMC(m)
-          expect_equal(length(grep("conjugate", conf$getSamplers()[[1]]$name)), 0,
-
-                       info = "incorrectly detected conjugacy")
-          code = nimbleCode({
-              for(i in 1:4) 
-                  y[i] ~ T(dnorm(mu[k[i]], sd = 1), 0, 1)
-              for(j in 1:5) 
-                  mu[j] ~ dnorm(0, 1)
-          })
-          m = nimbleModel(code, data = list(y = rnorm(4)),
-                          inits = list(k = rep(1,4)))
-          conf <- configureMCMC(m)
-          expect_equal(length(grep("conjugate", conf$getSamplers()[[1]]$name)), 0,
-                       info = "incorrectly detected conjugacy")
-          code = nimbleCode({
-              for(i in 1:4) 
-                  y[i] ~ dnorm(mu[k[i]], sd = 1)
-              for(j in 1:5) 
-                  mu[j] ~ T(dnorm(0, 1), 0 ,1)
-          })
-          m = nimbleModel(code, data = list(y = rnorm(4)),
-                          inits = list(k = rep(1,4)))
-          conf <- configureMCMC(m)
-          expect_equal(length(grep("conjugate", conf$getSamplers()[[1]]$name)), 0,
-                       info = "incorrectly detected conjugacy")
+
+    code = nimbleCode({
```

Comment:
nit: Ideally these could each go in a separate `test_that("...", {...})` so that they could fail independently. When I'm debugging, I find it really helpful to have as many as possible independent pass/fail variables to help me infer where the error might be.